frontend/src/models/task.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Initializes the shape for the data being retrieved from the backend.
+ */
+export interface Task {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+
+  // data received will be in JSON, JSON does not have a date type so we use
+  // a string instead
+  dateCreated: string;
```

Comment:
Not sure what "initializes" means here, this is just defining a type. I think we should solidify what the intended use of this type is though, in case it wasn't clear--this is what all the rest of the frontend should see when components use a Task object, not necessarily what our API client functions actually receive in an HTTP response. To that end, I'd change `dateCreated` back to a `Date` object and add the conversion logic in the API client. Since there will be multiple functions that return a Task, it probably makes sense to write some kind of `parseTask()` function which turns the output of `response.json()` into a `Task` object that frontend components can use, or throws an error if something doesn't match what's expected.